thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/diffsummary/costfunction/BalancedCostFunction.java

```diff
@@ -0,0 +1,108 @@
+package com.linkedin.thirdeye.client.diffsummary.costfunction;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import java.util.Map;
+
+public class BalancedCostFunction implements CostFunction {
```

Comment:
add java doc explaining the high level idea

Reply:
Done